quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/jaxrs/CloseSessionHandler.java

```diff
@@ -66,6 +66,9 @@ public void write(OutputStream output) throws IOException, WebApplicationExcepti
     }
 
     private void closeSession() {
-        close(Resteasy.getContextData(KeycloakSession.class));
+        KeycloakSession session = Resteasy.getContextData(KeycloakSession.class);
+        if (session != null && !DefaultKeycloakSession.class.cast(session).isClosed()) {
+            session.close();
+        }
```

Comment:
I'll just leave it as is for now, since that's what it has always been. The only reason to change it to an instanceof would be if we introduce some form of mock testing such that the instance isn't actually a DefaultKeyCloakSession.

Reply:
Works for me.